packages/api/src/controllers/clip.ts

```diff
@@ -61,49 +58,35 @@ app.use(
 );
 
 async function getProcessingClipsByRequesterId(
-  requesterId: string
-): Promise<WithID<Asset>[]> {
-  const createdAfter = Date.now() - 10 * 60 * 1000;
-  const assets = await db.asset.find([
-    sql`data->'status'->>'phase' IN ('waiting', 'processing', 'running')`,
-    sql`data->'source'->>'requesterId' = ${requesterId}`,
-    sql`coalesce((data->>'createdAt')::bigint, 0) > ${createdAfter}`,
+  requesterId: string,
+  ownerId: string
+): Promise<WithID<Task>[]> {
+  const scheduledAt = Date.now() - 5 * 60 * 1000;
+  const [tasks] = await db.task.find([
+    sql`data->>'userId' = ${ownerId}`,
+    sql`coalesce((data->>'scheduledAt')::bigint, 0) > ${scheduledAt}`,
   ]);
+  let runningClips: WithID<Task>[] = [];
+  const tasksByRequester = tasks.filter((t) => t.requesterId === requesterId);
+  tasksByRequester.forEach((task) => {
+    if (task.type === "clip") {
+      if (
+        task.status.phase === "running" ||
+        task.status.phase === "pending" ||
+        task.status.phase === "waiting"
```

Comment:
This should be filtered directly on the query. You might also be able to re-use [this index](https://github.com/livepeer/livepeer-com/blob/23b1aa453bbef83981ad2547169c7149f0436e35/packages/api/src/store/task-table.ts#L77) if you query by `updatedAt` instead of `scheduledAt` (and pending tasks are gonna be updated at least every minute)